lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -90,36 +158,43 @@ def create(self, subsection, course_structure, course):
         """
         Returns the SubsectionGrade object for the student and subsection.
         """
+        self._prefetch_scores(course_structure, course)
         return (
-            self._get_saved_grade(subsection, course) or
+            self._get_saved_grade(subsection, course_structure, course) or
             self._compute_and_update_grade(subsection, course_structure, course)
         )
 
     def _compute_and_update_grade(self, subsection, course_structure, course):
         """
         Freshly computes and updates the grade for the student and subsection.
         """
-        self._prefetch_scores(course_structure, course)
         subsection_grade = SubsectionGrade(subsection)
         subsection_grade.compute(self.student, course_structure, self._scores_client, self._submissions_scores)
         self._update_saved_grade(subsection_grade, subsection, course)
         return subsection_grade
 
-    def _get_saved_grade(self, subsection, course):  # pylint: disable=unused-argument
+    def _get_saved_grade(self, subsection, course_structure, course):  # pylint: disable=unused-argument
         """
-        Returns the saved grade for the given course and student.
+        Returns the saved grade for the student and subsection.
         """
         if settings.FEATURES.get('ENABLE_SUBSECTION_GRADES_SAVED') and course.enable_subsection_grades_saved:
-            # TODO Retrieve the saved grade for the subsection, if it exists.
-            pass
+            try:
+                model = PersistentSubsectionGrade.read_grade(
+                    user_id=self.student.id,
+                    usage_key=subsection.location,
+                )
+                subsection_grade = SubsectionGrade(subsection)
+                subsection_grade.load_from_data(model, course_structure, self._scores_client, self._submissions_scores)
+                return subsection_grade
+            except PersistentSubsectionGrade.DoesNotExist:
+                return None
 
     def _update_saved_grade(self, subsection_grade, subsection, course):  # pylint: disable=unused-argument
```

Comment:
Similarly, should this be `_save_grade` since the grade isn't being updated, just saved?